index.js

```diff
@@ -140,10 +137,12 @@ module.exports = class Hyperdrive extends ReadyResource {
     const blobsCore = this.corestore.get({
       key: blobsKey,
       cache: false,
-      onwait: this._onwait
+      onwait: this._onwait,
+      writable: !this._readonly
     })
     await blobsCore.ready()
 
+    // TODO: readonly related?
     if (this.closing) {
       await blobsCore.close()
       return false
```

Comment:
This condition was added in the middle of the `readonly` feature, is this still needed or related? Any hints on how to test it?

Reply:
thats needed in general yea, otherwise this session would leak